Changes in module 'nf-core/bowtie2/build'
--- modules/nf-core/bowtie2/build/main.nf
+++ modules/nf-core/bowtie2/build/main.nf
@@ -5,7 +5,7 @@
     conda "bioconda::bowtie2=2.4.4"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/bowtie2:2.4.4--py39hbb4e92a_0' :
-        'biocontainers/bowtie2:2.4.4--py39hbb4e92a_0' }"
+        'quay.io/biocontainers/bowtie2:2.4.4--py39hbb4e92a_0' }"
 
     input:
     tuple val(meta), path(fasta)

************************************************************
