Changes in module 'nf-core/diamond/makedb'
--- modules/nf-core/diamond/makedb/main.nf
+++ modules/nf-core/diamond/makedb/main.nf
@@ -5,7 +5,7 @@
     conda "bioconda::diamond=2.0.15"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/diamond:2.0.15--hb97b32f_0' :
-        'biocontainers/diamond:2.0.15--hb97b32f_0' }"
+        'quay.io/biocontainers/diamond:2.0.15--hb97b32f_0' }"
 
     input:
     path fasta

************************************************************
