Changes in module 'nf-core/kaiju/kaiju'
--- modules/nf-core/kaiju/kaiju/main.nf
+++ modules/nf-core/kaiju/kaiju/main.nf
@@ -1,6 +1,7 @@
 process KAIJU_KAIJU {
     tag "$meta.id"
     label 'process_high'
+    label 'process_high_memory'
 
     conda "bioconda::kaiju=1.8.2"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?

************************************************************
