Changes in module 'nf-core/kaiju/kaiju2table'
--- modules/nf-core/kaiju/kaiju2table/main.nf
+++ modules/nf-core/kaiju/kaiju2table/main.nf
@@ -5,7 +5,7 @@
     conda "bioconda::kaiju=1.8.2"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/kaiju:1.8.2--h5b5514e_1':
-        'biocontainers/kaiju:1.8.2--h2e03b76_0' }"
+        'quay.io/biocontainers/kaiju:1.8.2--h2e03b76_0' }"
 
     input:
     tuple val(meta), path(results)
@@ -24,7 +24,7 @@
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
     dbnodes=`find -L ${db} -name "*nodes.dmp"`
-    dbname=`find -L ${db} -name "*.fmi" -not -name "._*"`
+    dbname=`find -L ${db} -name "*names.dmp"`
     kaiju2table   $args \\
         -t \$dbnodes \\
         -n \$dbname \\

************************************************************
