Changes in module 'nf-core/diamond/blastx'
--- modules/nf-core/diamond/blastx/main.nf
+++ modules/nf-core/diamond/blastx/main.nf
@@ -1,11 +1,13 @@
 process DIAMOND_BLASTX {
     tag "$meta.id"
-    label 'process_medium'
+    label 'process_high'
+    label 'process_long'
+    label 'process_high_memory'
 
     conda "bioconda::diamond=2.0.15"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/diamond:2.0.15--hb97b32f_0' :
-        'biocontainers/diamond:2.0.15--hb97b32f_0' }"
+        'quay.io/biocontainers/diamond:2.0.15--hb97b32f_0' }"
 
     input:
     tuple val(meta), path(fasta)
@@ -16,7 +18,7 @@
     output:
     tuple val(meta), path('*.blast'), optional: true, emit: blast
     tuple val(meta), path('*.xml')  , optional: true, emit: xml
-    tuple val(meta), path('*.txt')  , optional: true, emit: txt
+    tuple val(meta), path('*.txt.gz')  , optional: true, emit: txt
     tuple val(meta), path('*.daa')  , optional: true, emit: daa
     tuple val(meta), path('*.sam')  , optional: true, emit: sam
     tuple val(meta), path('*.tsv')  , optional: true, emit: tsv
@@ -58,7 +60,7 @@
         --out ${prefix}.${out_ext} \\
         --log
 
-    mv diamond.log ${prefix}.log
+    mv diamond.log ${prefix}.diamond.log
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":

************************************************************
